src/rp2_common/pico_lwip/tools/makefsdata.py

```diff
@@ -53,9 +53,13 @@ def process_file(input_dir, file):
         comment = f"\"Content-Length: {file_size}\" ({len(data)} chars)"
         results.append({'data': bytes(data, "utf-8"), 'comment': comment});
 
-    # content type
-    data = f"Content-Type: {content_type}\r\n\r\n"
-    comment = f"\"Content-Type: {content_type}\" ({len(data)} chars)"
+    # content type and content encoding
+    if content_encoding is None:
+        data = f"Content-Type: {content_type}\r\n\r\n"
+        comment = f"\"Content-Type: {content_type}\" ({len(data)} chars)"
+    else:
+        data = f"Content-Type: {content_type}\r\nContent-Encoding: {content_encoding}\r\n\r\n"
+        comment = f"\"Content-Type: {content_type} Content-Encoding: {content_encoding}\" ({len(data)} chars)"
```

Comment:
Just a small (untested) tidy-up suggestion:
```suggestion
    content_type_header = f"Content-Type: {content_type}"
    if content_encoding is None:
        data = f"{content_type_header}\r\n\r\n"
        comment = f"\"{content_type_header}\" ({len(data)} chars)"
    else:
        content_encoding_header = f"Content-Encoding: {content_encoding}"
        data = f"{content_type_header}\r\n{content_encoding_header}\r\n\r\n"
        comment = f"\"{content_type_header} {content_encoding_header}\" ({len(data)} chars)"
```